luigi/task.py

```diff
@@ -647,12 +661,15 @@ def input(self):
 
     def deps(self):
         """
-        Internal method used by the scheduler.
-
         Returns the flattened list of requires.
+
+        Override in "template" tasks which themselves are supposed to be
+        subclassed and thus have their requires() overridden (name preserved to
+        provide consistent end-user experience), yet need to introduce
+        (non-input) dependencies.
         """
         # used by scheduler
-        return flatten(self._requires())
+        return flatten(self.process_requires(self.requires()))
```

Comment:
or this be `flatten(self.process_requires(flatten(self.requires())))`

Reply:
Assumption for process_requires is that it can handle any valid structured requirements and doesn't have any side effects like flattening the result. So `self.process_requires(self.requires())` should return a structure similar to what you expect from `self.requires()`. It's the job of `dep` flatten it as expected by it's users.